### PR TITLE
GH-1216: [TDR New Work Detection] Extend System Tests for COVID processing

### PR DESCRIPTION
**Pull request by tbl3rd**
### Purpose

- https://broadinstitute.atlassian.net/browse/GH-1216

### Changes

- Add a system test for Covid workloads.
- We might be able to test more now, but I want to get feedback on what's here already.

### Review Instructions

- The interesting changes are in v1_endpoint_test.clj, spec.clj, and util.clj.  Review them in that order.
- Also tweeked a few docstrings and reindented some lines for consistency and readability.
- And updated dependencies while debugging the route specs.

### CODE REVIEW

**Review by ehigham — APPROVED** [review]
Looks great. Thank you for making those improvements too.

File: api/src/wfl/api/spec.clj [hunk 1]
```diff
@@ -128,3 +103,92 @@
 (s/def ::workflow-request (s/keys :req-un [::end
                                            ::environment
                                            ::start]))
+
+(s/def ::batch-executor string?)
+(s/def ::covid-executor (s/keys :req-un [::fromSource
+                                         ::methodConfiguration
+                                         ::methodConfigurationVersion
+                                         ::workspace]
+                                :opt-un [::name]))
```
[the remark this answers is not in the document]
Comment (ehigham) [reply api/src/wfl/api/spec.clj:112]: Likewise it's not included in the source and sink specs too

[the remark this answers is not in the document]
Comment (tbl3rd) [reply api/src/wfl/api/spec.clj:112]: I thought the name was the user label.  Will req it.

File: api/src/wfl/api/spec.clj [hunk 2]
```diff
@@ -128,3 +103,92 @@
 (s/def ::workflow-request (s/keys :req-un [::end
                                            ::environment
```
[the remark this answers is not in the document]
Comment (tbl3rd) [reply api/src/wfl/api/spec.clj:104]: OK.

File: api/src/wfl/api/spec.clj [hunk 3]
```diff
@@ -128,3 +103,92 @@
...
+
+(s/def ::workload-request (s/or :batch ::batch-workload-request
+                                :covid ::covid-workload-request))
+
+(s/def ::workload-response (s/or :batch ::batch-workload-response
+                                 :covid ::covid-workload-response))
+
+(s/def ::workload-responses (s/* ::workload-response))
+
+(s/def ::whatever (constantly true))
```
[the remark this answers is not in the document]
Comment (tbl3rd) [reply api/src/wfl/api/spec.clj:194]: Thanks.  A debugging hack I forgot to remove.

File: api/src/wfl/api/spec.clj [hunk 4]
```diff
@@ -128,3 +103,92 @@
...
+                        :covid ::covid-executor))
+
+(s/def ::column string?)
+(s/def ::dataset string?)
+(s/def ::entity string?)
+(s/def ::fromOutputs map?)
+(s/def ::fromSource string?)
+(s/def ::labels (s/* string?))
+(s/def ::name string?)
+(s/def ::methodConfiguration string?)
```
[the remark this answers is not in the document]
Comment (tbl3rd) [reply api/src/wfl/api/spec.clj:124]: Good idea.

**Pull request merged**